lib/generation/generators/middleware/state_management/utils/middleware_utils.dart

```diff
@@ -77,16 +77,18 @@ class MiddlewareUtils {
       ${manager.generateImports()}
       class ${defaultStateName} extends ChangeNotifier {
 
-      Widget currentWidget;
-      ${defaultStateName}(){}
-
-      // default provider event handler for gestures.
-      void onGesture() {
-      }
+        LayoutBuilder currentLayout;
+        final String widgetName;
+        ${defaultStateName}(this.widgetName);
```

Comment:
Hi @GVTopCoder, I'm currently reviewing this PR. Is there a reason for each `model` having a `widgetName`? Currently I don't see it being utilized anywhere 🤔